server/util/claims/claims.go

```diff
@@ -141,14 +141,18 @@ func ParseClaims(token string) (*Claims, error) {
 }
 
 func APIKeyGroupClaims(akg interfaces.APIKeyGroup) *Claims {
+	keyRole := role.Default
+	// User management through SCIM requires Admin access.
+	if akg.GetCapabilities()&int32(akpb.ApiKey_SCIM_CAPABILITY) > 0 {
+		keyRole = role.Admin
+	}
```

Comment:
Instead of promoting the SCIM key to admin here, could `GetUserByID` check that the user is either an admin OR the currently authenticated API key has SCIM capability?

Reply:
If we do that, we'd have to do this for every other management function we call from the SCIM code. 
I think we should avoid carving out exceptions for specific API key types. 
The SCIM use case is the first use case where an API key is used to perform administrative operations and in this case I think treating the API key as having an admin role is appropriate.

@tylerwilliams for second opinion

Reply:
I think keeping it simple here, like @vadimberezniker has suggested, is probably the right move until we have a stronger reason for doing it differently. (Theres a good chance all of this scim stuff will change in the future anyway)